web/app/js/components/util/MetricUtils.jsx

```diff
@@ -171,6 +176,14 @@ export const processMultiResourceRollup = rawMetrics => {
       return;
     }
 
+    // the Stat API returns trafficsplit data if queried with resource_type=all.
+    // trafficsplit leaves cannot be "upstream" or "downstream" of other
+    // resources, so we do not process trafficsplit data if the requested
+    // resourceType is not "trafficsplit" or "all".
+    if (resourceType !== "all" && resourceType !== "trafficsplit" && table.podGroup.rows[0].tsStats) {
+      return;
```

Comment:
are we returning early here because the presence of `table.podGroup.rows[0].tsStats` implies that this row only has trafficsplit data and nothing else (`BasicStats` or `TcpStats`)? if so, mind adding that to the comment, and consider moving the `table.podGroup.rows[0].tsStats` to the front of the `if`, to signify its significance? or, if i'm missing something else, please enlighten me. =)

Reply:
Great question. Because trafficsplit metrics are now being returned from the Stat API endpoint, the current `processMultiResourceRollup` can erroneously return trafficsplit leaf services as "upstream" or "downstream" of a selected resource in the Octopus graph.

This line checks if the statTable contains trafficsplit data and returns early if the selected resource is not "all" or "trafficsplit", since any other resource should not include trafficsplit metrics.

I updated the comment to reflect this. The better longterm solution would be to refactor `processMultiResourceRollup` so that `upstream` and `downstream` are handled separately. I'd like to do that in a separate PR.